crates/sui-storage/src/lock_service.rs

```diff
@@ -114,8 +123,8 @@ impl LockServiceImpl {
     /// Returns the state of a single lock.
     /// * None - lock does not exist and is not initialized
     /// * Some(None) - lock exists and is initialized, but not locked to a particular transaction
-    /// * Some(Some(tx_digest)) - lock exists and set to transaction
-    fn get_lock(&self, object: ObjectRef) -> Result<Option<Option<TransactionDigest>>, SuiError> {
+    /// * Some(Some(lock_info)) - lock exists and set to transaction from a specific epoch.
+    fn get_lock(&self, object: ObjectRef) -> Result<Option<Option<LockInfo>>, SuiError> {
         self.transaction_lock
             .get(&object)
```

Comment:
Should we try to find out the lock we are trying to take is overridden with a newer epoch and fail very loudly? It is like a system invariant and I'd assume it should almost never happen but if it does we will be able to debug it better.

Reply:
I would agree with Sadhan, though here in the lock service Is not where such a check should take place.

Reply:
hmm how do we know, though?

Reply:
I realize we need to better define what the scenario is.